src/CameraParameters.h

```diff
@@ -0,0 +1,189 @@
+    static const char PIXEL_FORMAT_YUV420P[];
+    static const char PIXEL_FORMAT_RGB565[];
+    static const char PIXEL_FORMAT_RGBA8888[];
+    // Values for Image formats
+    static const char IMAGE_FORMAT_JPEG[];
+    static const char IMAGE_FORMAT_PNG[];
+
+    enum param_id {
+        PARAM_ID_CAMERA_MODE = 1,
+        PARAM_ID_BRIGHTNESS,
+        PARAM_ID_CONTRAST,
+        PARAM_ID_SATURATION,
+        PARAM_ID_HUE,
+        PARAM_ID_WHITE_BALANCE_MODE,
+        PARAM_ID_GAMMA,
+        PARAM_ID_GAIN,
+        PARAM_ID_POWER_LINE_FREQ_MODE,
+        PARAM_ID_WHITE_BALANCE_TEMPERATURE,
+        PARAM_ID_SHARPNESS,
+        PARAM_ID_BACKLIGHT_COMPENSATION,
+        PARAM_ID_EXPOSURE_MODE,
+        PARAM_ID_EXPOSURE_ABSOLUTE,
+        PARAM_ID_IMAGE_SIZE,
+        PARAM_ID_IMAGE_FORMAT,
+        PARAM_ID_PIXEL_FORMAT,
+        PARAM_ID_SCENE_MODE,
+        PARAM_ID_VIDEO_SIZE,
+        PARAM_ID_VIDEO_FRAME_FORMAT,
+        PARAM_ID_VIDEO_SNAPSHOT_SUPPORTED
+    };
+
+    enum id_image_size {
+        ID_IMAGE_SIZE_3264x2448 = 0,
+        ID_IMAGE_SIZE_3264x1836,
+        ID_IMAGE_SIZE_1920x1080
+    };
+
+    enum id_video_size {
+        ID_VIDEO_SIZE_1920x1080x30 = 1,
+        ID_VIDEO_SIZE_1920x1080x15,
+        ID_VIDEO_SIZE_1280x720x30,
+        ID_VIDEO_SIZE_1280x720x15,
+        ID_VIDEO_SIZE_960x540x30,
+        ID_VIDEO_SIZE_960x540x15,
+        ID_VIDEO_SIZE_848x480x30,
+        ID_VIDEO_SIZE_848x480x15,
+        ID_VIDEO_SIZE_640x480x60,
+        ID_VIDEO_SIZE_640x480x30,
+        ID_VIDEO_SIZE_640x480x15,
+        ID_VIDEO_SIZE_640x360x30,
+        ID_VIDEO_SIZE_640x360x15,
+        ID_VIDEO_SIZE_424x240x30,
+        ID_VIDEO_SIZE_424x240x15,
+        ID_VIDEO_SIZE_320x240x60,
+        ID_VIDEO_SIZE_320x240x30,
+        ID_VIDEO_SIZE_320x240x15,
+        ID_VIDEO_SIZE_320x180x30,
+        ID_VIDEO_SIZE_320x180x60
+    };
+
+    enum id_camera_mode { ID_CAMERA_MODE_STILL = 0, ID_CAMERA_MODE_VIDEO, ID_CAMERA_MODE_PREVIEW };
+
+    enum id_image_format { ID_IMAGE_FORMAT_JPEG = 0, ID_IMAGE_FORMAT_PNG };
+
+    enum id_white_balance_mode {
+        ID_WHITE_BALANCE_MANUAL = 0,
+        ID_WHITE_BALANCE_AUTO,
+        ID_WHITE_BALANCE_INCANDESCENT,
+        ID_WHITE_BALANCE_FLUORESCENT,
+        ID_WHITE_BALANCE_WARM_FLUORESCENT,
+        ID_WHITE_BALANCE_DAYLIGHT,
+        ID_WHITE_BALANCE_CLOUDY_DAYLIGHT,
+        ID_WHITE_BALANCE_TWILIGHT,
+        ID_WHITE_BALANCE_SHADE
+    };
+
+    enum id_exposure_mode { ID_EXPOSURE_MANUAL = 0, ID_EXPOSURE_AUTO };
+
+    enum id_pixel_format {
+        ID_PIXEL_FORMAT_YUV422SP = 0,
+        ID_PIXEL_FORMAT_YUV420SP,
+        ID_PIXEL_FORMAT_YUV422I,
+        ID_PIXEL_FORMAT_YUV420P,
+        ID_PIXEL_FORMAT_RGB565,
+        ID_PIXEL_FORMAT_RGBA8888
+    };
+
+    // TODO :: Make exhaustive list of parameters and its possible values
+private:
+    void initParamIdType();
+    void set(std::map<std::string, std::string> &pMap, std::string key, std::string value);
```

Comment:
Done. I initially created function to be a placeholder for parameter validity checks etc but seems that would be redundant.